Update modules content overflow to auto

renatho opened the pull request:
### Changes proposed in this Pull Request

* Update overflow to auto. It allows the user to scroll into the module if there are many lessons.
* If we don't want this scroll when there are many lessons, we could change the animation for something that doesn't depend on the `max-height` (scale/opacity/translate).

### Testing instructions

* Add many lessons to a module until to be more than `2000px` height.
* Make sure you can scroll through the content.
* Collapse and expand the module box, and make sure the animation still works properly.

yscik commented:
This would add a scrollbar flashing up during the animation:

![QuickTime movie](https://user-images.githubusercontent.com/176949/94284601-e86bc700-ff52-11ea-9981-ffc79f42fa76.gif)

What if remove `max-height` from CSS and add a call to set the `max-height` to the actual content height (or just a bit more to be safe) in jQuery?

renatho commented:
Thank you for checking it, @yscik!

> This would add a scrollbar flashing up during the animation:

Good catch! I had checked only in Chrome.

> What if remove `max-height` from CSS and add a call to set the `max-height` to the actual content height (or just a bit more to be safe) in jQuery?

Yep! I'm not sure if we'll have a different solution for that. I tested some other animations only changing CSS now, but they didn't look cool. I'll let to test it with a little more time.

renatho commented:
A new suggestion:

![Sep-25-2020 15-20-22](https://user-images.githubusercontent.com/876340/94303708-a3d92f00-ff44-11ea-90f5-fde011e22cea.gif)

The bad side of this solution is that the collapse will be suddenly, and the expand, animated. Although I think it doesn't bother, and it's a clean solution (only CSS).

It'll animate when opening the page too.

yscik commented:
I don't think we need for worry about a CSS-only solution here, since the state is toggled by Javascript anyway.
I like the opacity fade, but we could do some more for the height.

What if we just use jQuery's animate? This would only need an `overflow: hidden` in CSS:
```js
moduleDetails.animate( { height: collapsed ? 0 : moduleDetails.prop( 'scrollHeight' ) }, 300 );
```

gkaragia commented:
> What if we just use jQuery's animate? This would only need an `overflow: hidden` in CSS:

Not sure if it is an issue but won't this add jQuery as a requirement to customer sites?

renatho commented:
> What if we just use jQuery's animate? This would only need an overflow: hidden in CSS:

My most concern about that is adding side-effects to react components (On the editor side).

> On the other hand we can conditionally enqueue it when the animations are enabled.

And I don't think it worths add the jQuery only for this animation. But if we really want that, maybe we could add a Vanilla solution for the frontend, and keep it without animation in the editor. I think it's an acceptable exception.

But I still would prefer a cleaner solution only with the cons to not animate while collapsing. In my opinion, it's also nice.

yscik commented:
> Not sure if it is an issue but won't this add jQuery as a requirement to customer sites?

The script for this is already written with jQuery, so it's needed, and it was also used for other Sensei frontend JS.

We could also consider moving away from jQuery to something lighter. It's not used in many places on the frontend yet. Switching to vanilla JS is an option, or using a light framework like Preact or Vue? Maybe we can have some discussions in a Dev Roundtable or some experiments?

renatho commented:
Refactored a little, using different approaches for the animation in the frontend and in the editor.

I also took the opportunity to get rid of the `jQuery` in this part and I changed the arrow to a button. 😉

yscik commented on includes/blocks/class-sensei-course-outline-block.php:
```diff
@@ -251,9 +251,11 @@ private function render_module_block( $block, $course_id, $outline_attributes )
+					</button>
 				</header>
-				<div class="wp-block-sensei-lms-collapsible ' . $animated . '">
+				<div class="wp-block-sensei-lms-collapsible expanded ' . $animated . '">
```
I think this is not needed anymore.

renatho replied:
Good catch! The class and the jQuery were removed here:
https://github.com/Automattic/sensei/pull/3646/commits/47b1beacb49b38a40ebe5cee919841a374efeb1e https://github.com/Automattic/sensei/pull/3646/commits/bc329629a58d02abfb23e8bbc820f797b7d93265

yscik reviewed (commented):
Looks good!

We should also remove the `jquery` dependency from the script enqueue call here:
https://github.com/Automattic/sensei/blob/5e8e3b8f0e27ee37f7983e668d1ff8c55729df97/includes/class-sensei-frontend.php#L147

yscik reviewed (dismissed):
All good! ✈️

renatho commented:
I also took the opportunity to move the enqueue script to the course outline block logic. https://github.com/Automattic/sensei/pull/3646/commits/825e3fb1db0fc4faee41c5a9325f87a49f1b9427

yscik reviewed (approved):
👌
